File: app/src/main/java/com/example/win/easy/dagger/module/LoginModule.java
package com.example.win.easy.dagger.module;

import com.example.win.easy.web.request.BackendRequestService;
import com.example.win.easy.web.service.LoginService;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class LoginModule {

    @Provides
    @Singleton
    static LoginService providesLoginService(BackendRequestService backendRequestService){
        return new LoginService(backendRequestService);
    }

}

File: app/src/main/java/com/example/win/easy/dagger/module/LoginModule.kt
package com.example.win.easy.dagger.module

import com.example.win.easy.network.BackendRequestService
import com.example.win.easy.network.LoginService
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
object LoginModule {

    @JvmStatic
    @Provides
    @Singleton
    fun providesLoginService(backendRequestService: BackendRequestService): LoginService = LoginService(backendRequestService)
}